ablySpec/RestChannel.swift

```diff
@@ -198,14 +222,228 @@ class RestChannel: QuickSpec {
                         
                         expect(message.content()).toNot(beNil())
                         expect(message.action).to(equal(ARTPresenceAction.Present))
-                        
+
                         // skip the encrypted message for now
-                        if message.payload.encoding?.rangeOfString("cipher") == nil {
+                        if message.payload.encoding.rangeOfString("cipher") == nil {
                             expect(message.content() as? NSObject).to(equal(fixtureMessage["data"].object as? NSObject))
                         }
                     }
                 }
             }
         }
+
+        describe("message encoding") {
+
+            struct TestCase {
+                let value: AnyObject?
+                let expected: JSON
+            }
+
+            let text = "John"
+            let integer = "5"
+            let decimal = "65.33"
+            let dictionary = ["number":3, "name":"John"]
+            let array = ["John", "Mary"]
+            let binaryData = NSString(string: "123456").dataUsingEncoding(NSUTF8StringEncoding)!
+
+            // RSL4a
+            it("payloads should be binary, strings, or objects capable of JSON representation") {
+                let validCases = [
+                    TestCase(value: nil, expected: JSON([:])),
+                    TestCase(value: text, expected: JSON(["data": text])),
+                    TestCase(value: integer, expected: JSON(["data": integer])),
+                    TestCase(value: decimal, expected: JSON(["data": decimal])),
+                    TestCase(value: dictionary, expected: JSON(["data": dictionary, "encoding": "json"])),
+                    TestCase(value: array, expected: JSON(["data": array, "encoding": "json"])),
+                    TestCase(value: binaryData, expected: JSON(["data": binaryData.toBase64, "encoding": "base64"])),
+                ]
+
+                client.httpExecutor = mockExecutor
+
+                validCases.forEach { caseTest in
+                    waitUntil(timeout: testTimeout) { done in
+                        channel.publish(caseTest.value, callback: { error in
+                            expect(error).to(beNil())
+
+                            if let request = mockExecutor.requests.last,
```

Comment:
When there's no chance that an optional is nil, they should be unwrapped with `!` instead of if-let-ed. That way, if by any chance it _is_ nil (say we change and break MockHTTPExecutor by accident), this doesn't fail silently.

This would also fail silently if `request.HTTPBody` is nil, but in that case a guard-let would be a better solution, as we've talked about before, because that's actually tested code.